src/core/unit.c

```diff
@@ -3050,6 +3050,9 @@ int unit_add_dependency(
                 return 0;
         }
 
+        if (u && u->manager && FLAGS_SET(u->manager->test_run_flags, MANAGER_TEST_RUN_IGNORE_DEPENDENCIES))
```

Comment:
The comment above says "default dependencies", but this check is done early enough to cause all dependencies to be ignored. The commit message implies something different yet. I don't know what is intended. (This is compounded by the almost complete lack of descriptions in commit messages: the reader is left guessing.)